bitcoin/src/network.rs

```diff
@@ -171,13 +233,25 @@ impl Network {
 
     /// Returns the associated network parameters.
     pub const fn params(self) -> &'static Params {
-        const PARAMS: [Params; 4] = [
+        const PARAMS: [Params; 5] = [
             Params::new(Network::Bitcoin),
-            Params::new(Network::Testnet),
+            Params::new(Network::Testnet(TestnetVersion::V3)),
+            Params::new(Network::Testnet(TestnetVersion::V4)),
             Params::new(Network::Signet),
             Params::new(Network::Regtest),
         ];
-        &PARAMS[self as usize]
+        &PARAMS[self.as_index()]
+    }
+
+    /// Returns the associated network index.
+    pub const fn as_index(self) -> usize {
```

Comment:
Don't make this `pub`, it'll just trigger panics in user code.

Reply:
Yes, I have also tweaked this so that this internal API should not be pub